### PR TITLE
Clean lan91c111 netcard

### PR DESCRIPTION
**Pull request by anton-bondarev**
* Clean lan91c111 netcard
* Enable deferred sending if ALLOC_TX was not succed

### CODE REVIEW

File: src/drivers/net/lan91c111/lan91c111.c [hunk 1]
```diff
@@ -252,89 +204,127 @@ static const struct net_driver lan91c111_drv_ops = {
...
+	struct smc_local *smc;
+	int ret;
+
+	smc = netdev_priv(dev);
+
+	skb = skb_queue_pop(&smc->pending_queue);
+	if (!skb) {
+		return 0;
+	}
+	lan91c111_send_hw(skb);
```
Comment (Deryugin) [inline src/drivers/net/lan91c111/lan91c111.c:218]: Looks like theese lines can be removed as `while` look does the same thing.

File: src/drivers/net/lan91c111/lan91c111.c [hunk 2]
```diff
@@ -252,89 +204,127 @@ static const struct net_driver lan91c111_drv_ops = {
...
-		skb_data = skb_data_cast_in(skb->data);
-		assert(skb_data);
+	lan91c111_set_cmd(CMD_RX_POP_AND_RELEASE);
 
-		for (i = 0; i < len >> 2; i++) {
-			buf = REG32_LOAD(BANK_DATA);
-			*((uint32_t *)(skb_data + i * 4)) = buf;
-		}
+	if (buf & (ODD_CONTROL << 8)) {
+		skb->len++;
```
Comment (Deryugin) [inline src/drivers/net/lan91c111/lan91c111.c:274]: Are you sure it's neccessary? This also may cause buffer overflow.

File: src/drivers/net/lan91c111/lan91c111.c [hunk 3]
```diff
@@ -164,55 +78,93 @@ static void lan91c111_push_data(uint8_t *data, int len) {
...
+
+static int lan91c111_add_pending(struct smc_local *smc, struct sk_buff *skb) {
+	uint8_t int_mask;
+
+	lan91c111_set_bank(2);
+
+	skb_queue_push(&smc->pending_queue, skb);
+
+	int_mask = REG8_LOAD(BANK_INTERRUPT_MASK);
+	int_mask &= ~ALLOC_INT;
```
Comment (Deryugin) [inline src/drivers/net/lan91c111/lan91c111.c:122]: Should it be `int_mask |= ALLOC_INT` instead to enable irq?

File: src/drivers/net/lan91c111/lan91c111.c (hunk 1 above)
Comment (anton-bondarev) [reply src/drivers/net/lan91c111/lan91c111.c:218]: Yes, but, when we first-time appear in this point it means that ALLOC_TX command was finished. Of cause we can call ALLOC_TX again, it's not significant I think

File: src/drivers/net/lan91c111/lan91c111.c (hunk 2 above)
Comment (anton-bondarev) [reply src/drivers/net/lan91c111/lan91c111.c:274]: this code just move from old file line 315 to own routine lan91c111_recv_hw

File: src/drivers/net/lan91c111/lan91c111.c (hunk 3 above)
Comment (anton-bondarev) [reply src/drivers/net/lan91c111/lan91c111.c:122]: ooh, thanks a lot.
Done!

**Pull request merged**